### DISCUSSION TITLE
Confirmation before executing a shell command

### DISCUSSION THREAD

**Discussion by Taitava** [Taitava/obsidian-shellcommands#27] [Ideas - done & released]
Some commands can be quite dangerous, especially if they involve deleting files, writing to files, or use unpredictable variables such as `{{selection}}` or `{{clipboard}}` ([there is no way to escape/encode variable values at the moment](https://github.com/Taitava/obsidian-shellcommands/issues/11)).

I'd like to add a simple toggle for each shell command. If the toggle is on, then every time the user tries to execute the shell command, they are prompted like this:

> ## Actual shell command here
> **Execute this shell command?**
> Alias: command alias here
> [Yes, execute] (no cancel button, user can click X to cancel, or click background or hit Esc)

Later I could add an option to only confirm if the command is executed via a certain way. E.g. executing via command palette does not require confirmation (because a user can see variables in the command palette), but executing via a hotkey requires a confirmation. This could decided on a per command basis.

**Comment by FelipeRearden** [comment]
I think its a great idea @Taitava !

One more feature to the `cog icon` !

**Comment by Taitava** [comment]
Nice. Btw does it need a description field too? For describing how the command can be dangerous. But maybe it's better to start without a description field, as that would fill the UI space in the *cog modal*/*alias modal*/whatever we should call the popup window.

**Reply by FelipeRearden** [reply]
I think there is not need of a description field too. 

Let’s save some space for the future features :)

**Comment by Taitava** [comment]
I created now issue #35 for this. I'm leaving out:
 - Description field
 - > Later I could add an option to only confirm if the command is executed via a certain way

These can be implemented later, in separate issues, but they can still be discussed here.
